File: SPCode/CS/Data/Support/SPServices.cs
using System.Text.RegularExpressions;
using System.Diagnostics;
using System;
using System.Drawing.Drawing2D;
using System.Windows.Forms;
using System.Collections;
using System.Drawing;
using Microsoft.VisualBasic;
using System.Data.SqlClient;
using System.Data;
using System.Collections.Generic;
using WeifenLuo.WinFormsUI;
using Microsoft.Win32;
using WeifenLuo;
//using SoftLogik.Win.SPDataProxy;
using SoftLogik.Win.SPDataProxyTableAdapters;


namespace SoftLogik.Win
{
	public class SPServices
	{
		
		
		public static string GenerateAutoID(string Prefix)
		{
			SPAutoIdDataTable data;
			SPAutoIdRow retData;
			using (SoftLogik.Win.SPDataProxyTableAdapters.taSPAutoId dataCmd = new SoftLogik.Win.SPDataProxyTableAdapters.taSPAutoId())
			{
				data = dataCmd.GetAutoID(Prefix, DateTime.Now);
			}
			
			retData = (SPAutoIdRow) (data.Rows[0]);
			return retData.NewID;
		}
		
	}
	
}

File: SPCode/Data/Support/SPServices.vb
Imports SoftLogik.Win.SPDataProxy
Imports SoftLogik.Win.SPDataProxyTableAdapters

Public Class SPServices

    Public Shared Function GenerateAutoID(ByVal Prefix As String) As String
        Dim data As SPAutoIdDataTable, retData As SPAutoIdRow
        Using dataCmd As New taSPAutoId
            data = dataCmd.GetAutoID(Prefix, Now)
        End Using
        retData = CType(data.Rows(0), SPAutoIdRow)
        Return retData.NewID
    End Function

End Class
